sodasql/dialects/redshift_spectrum_dialect.py

```diff
@@ -0,0 +1,31 @@
+#  Copyright 2020 Soda
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#   http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from sodasql.dialects.redshift_dialect import RedshiftDialect
+from sodasql.scan.dialect import REDSHIFT_SPECTRUM
+from sodasql.scan.parser import Parser
+
+
+class RedshiftSpectrumDialect(RedshiftDialect):
```

Comment:
I would personally make a switch in `RedshiftDialect` instead of making a new dialect. New dialect means separate testing pipelines jobs in soda-sql, maintenance burden across the entire platform. A simple boolean toggle `useRedshift` inside the Redshift setup might just do the trick and be good enough

Reply:
I thought of making it a switch, but wanted to start with a clean slate, because I didn’t fully analyze the SQL limitations. 

FYI: this PR is still a draft not ready for review yet 😊